### PR TITLE
All path.

### PR DESCRIPTION
**Pull request by CPWstatic**

### CODE REVIEW

**Review by nevermore3 — DISMISSED** [review]

File: src/executor/query/DataCollectExecutor.cpp [hunk 1]
```diff
@@ -155,5 +159,32 @@ Status DataCollectExecutor::collectBFSShortest(const std::vector<std::string>& v
     // Will rewrite this method once we implement returning the props for the path.
     return rowBasedMove(vars);
 }
+
+Status DataCollectExecutor::collectAllPaths(const std::vector<std::string>& vars) {
+    DataSet ds;
+    ds.colNames = std::move(colNames_);
+    DCHECK(!ds.colNames.empty());
+    // itersHolder keep life cycle of iters util this method return.
+    std::vector<std::unique_ptr<Iterator>> itersHolder;
```
Comment (laura-ding) [inline src/executor/query/DataCollectExecutor.cpp:168]: I don't understand. What's the use of `itersHolder` here?

File: src/validator/FindPathValidator.cpp [hunk 2]
```diff
@@ -142,5 +145,82 @@ GetNeighbors::EdgeProps FindPathValidator::buildEdgeKey(bool reverse) {
...
+
+    DCHECK(!!vids);
+    auto* gn = GetNeighbors::make(qctx_, dep, space_.id);
+    gn->setSrc(vids);
+    gn->setEdgeProps(buildEdgeKey(reverse));
+    gn->setInputVar(startVidsVar);
+
+    auto* allPaths = ProduceAllPaths::make(qctx_, gn);
+    allPaths->setInputVar(gn->outputVar());
+    allPaths->setColNames({"_vid", "path"});
```
Comment (laura-ding) [inline src/validator/FindPathValidator.cpp:190]: use kVid

File: src/validator/FindPathValidator.cpp [hunk 3]
```diff
@@ -142,5 +145,82 @@ GetNeighbors::EdgeProps FindPathValidator::buildEdgeKey(bool reverse) {
...
+    allPaths->setOutputVar(startVidsVar);
+
+    DataSet ds;
+    ds.colNames = {"_vid", "path"};
```
Comment (laura-ding) [inline src/validator/FindPathValidator.cpp:194]: ditto can make the colName as the one define

File: src/validator/FindPathValidator.cpp [hunk 4]
```diff
@@ -142,5 +145,82 @@ GetNeighbors::EdgeProps FindPathValidator::buildEdgeKey(bool reverse) {
...
+    for (auto& vid : starts.vids) {
+        Row row;
+        row.values.emplace_back(vid);
+
+        List paths;
+        Path path;
+        path.src = Vertex(vid.getStr(), {});
```
Comment (laura-ding) [inline src/validator/FindPathValidator.cpp:201]: vid must string?

Comment (CPWstatic) [reply src/validator/FindPathValidator.cpp:201]: Yes, string is the only option now for `Vertex`.

**Review by laura-ding — APPROVED** [review]

**Review by nevermore3 — APPROVED** [review]

**Pull request merged**